test-harness/index.ts

```diff
@@ -43,60 +45,79 @@ describe('harness', () => {
-                expect(res).toStrictEqual([]);
-                delete expected.body;
-                delete output.body;
+      const clientCommandHandle = spawnSync(command, args, {
+        shell: true,
+        encoding: 'utf8',
+        windowsVerbatimArguments: false,
+      });
+      const output: any = parseResponse(clientCommandHandle.stdout.trim());
+      const expected: any = parseResponse((parsed.expect || parsed.expectLoose).trim());
 
-                const isValid = validate(expected, output).valid;
-                expect(isValid).toBeTruthy();
+      const isXml = xmlValidator.test(
+        get(output, ['header', 'content-type'], ''),
+        expected.body
+      );
 
-                return shutdownPrism(prismMockProcessHandle, done);
-              });
-            }
+      try {
+        if (isXml) {
+          return xmlValidator.validate(expected, output).then(res => {
+            expect(res).toStrictEqual([]);
+            delete expected.body;
+            delete output.body;
 
             const isValid = validate(expected, output).valid;
+            expect(isValid).toBeTruthy();
 
-            if (!!isValid) {
-              expect(isValid).toBeTruthy();
-            } else {
-              expect(output).toMatchObject(expected);
-            }
-            if (parsed.expect) {
-              expect(output.body).toStrictEqual(expected.body);
-            }
-          } finally {
-            shutdownPrism(prismMockProcessHandle, done);
-          }
+            return shutdownPrism(prismMockProcessHandle, done);
+          });
         }
-      });
+
+        const isValid = validate(expected, output).valid;
+
+        if (!!isValid) {
+          expect(isValid).toBeTruthy();
+        } else {
+          expect(output).toMatchObject(expected);
+        }
+        if (parsed.expect) {
+          expect(output.body).toStrictEqual(expected.body);
+        }
+      } finally {
+        shutdownPrism(prismMockProcessHandle, done);
+      }
     });
   });
 });
 
-function shutdownPrism(processHandle: ChildProcess, done: jest.DoneCallback) {
+async function startPrism(server, filename): Promise<ChildProcess> {
+  return new Promise((resolve, reject) => {
+    const serverArgs = server.split(/ +/).map(t => t.trim().replace('${document}', filename));
+    const prismMockProcessHandle = spawn(path.join(__dirname, '../cli-binaries/prism-cli'), serverArgs);
+
+    const timeout = setTimeout(
+      () => {
+        shutdownPrism(prismMockProcessHandle, () => {
+          reject(new Error(`Timeout while waiting for "${WAIT_FOR_LINE}" log line`));
+        });
+      },
+      WAIT_FOR_LINE_TIMEOUT
+    );
+
+    prismMockProcessHandle.stdout.pipe(split2()).on('data', (line: string) => {
+      if (line.includes(WAIT_FOR_LINE)) {
+        clearTimeout(timeout);
+        resolve(prismMockProcessHandle);
+      }
+    });
+  });
+}
+
+function shutdownPrism(processHandle: ChildProcess, done: (...args: any[]) => any) {
```

Comment:
```suggestion
function shutdownPrism(processHandle: ChildProcess, done: jest.DoneCallback) {
```

Reply:
It has to stay that way to be compatible both with jest call and our call (https://github.com/stoplightio/prism/pull/825/files#diff-d26ad5d0c95e0973686e7fa034e1283bR107).

Reply:
Ah yeah ok, got it.